src/store.ts

```diff
@@ -871,25 +871,46 @@ export default class IndexedFormula extends Formula { // IN future - allow pass
   }
 
   /**
-   * Removes all statements in a doc, along with the related metadata including request/response
+   * Removes all metadata
    * @param doc - The document / graph
    */
-  removeDocument(doc: Quad_Graph): IndexedFormula {
-    const meta = this.sym('chrome://TheCurrentSession') // or this.rdfFactory.namedNode('chrome://TheCurrentSession')
-    const linkNamespaceURI = 'http://www.w3.org/2007/ont/link#' // alain
-    // remove request/response and metadata
-    const requests = this.statementsMatching(undefined, this.sym(`${linkNamespaceURI}requestedURI`), this.rdfFactory.literal(doc.value), meta).map(st => st.subject)
+  removeMetadata(doc: Quad_Graph): IndexedFormula {
+    const meta = this.fetcher?.appNode // this.sym('chrome://TheCurrentSession')
+    const linkNamespaceURI = 'http://www.w3.org/2007/ont/link#'
+    const kb = this
+    // removeMatches() --> removeMany() --> remove() fails on Collection
```

Comment:
Wouldn't it be better to fix the remove() function to work with Collection instead of applying a workarround here?

Reply:
Yes it would. I did not find howto.

Reply:
What goes wrong with when one f the nodes being removed is a collection?

Reply:
@timbl
This is I hope a detailed enough issue https://github.com/linkeddata/rdflib.js/issues/631